cocotb/_build_libs.py

```diff
@@ -0,0 +1,438 @@
+
+        # icarus requires vpl extension, gpivpi is default in Makfiles
+        if filename.find("icarus") >= 0:
+            filename_short = filename_short.replace("libvpi.so", "gpivpi.vpl")
+            filename_short = filename_short.replace("libvpi.dll", "gpivpi.vpl")
+
+        return filename_short
+
+    # Add extra library_dirs path
+    def finalize_options(self):
+        _build_ext.finalize_options(self)
+
+        for ext in self.extensions:
+            ext.library_dirs.append(
+                os.path.join(self.build_lib, os.path.dirname(ext._full_name))
+            )
+
+    # copy libs into proper directory in develop
+    def copy_extensions_to_source(self):
+        build_py = self.get_finalized_command("build_py")
+        for ext in self.extensions:
+            fullname = self.get_ext_fullname(ext.name)
+            filename = self.get_ext_filename(fullname)
+            modpath = fullname.split(".")
+            package = ".".join(modpath[:-1])
+            package_dir = build_py.get_package_dir(package)
+            dest_filename = os.path.join(package_dir, filename)
+            src_filename = os.path.join(self.build_lib, filename)
+
+            os.makedirs(os.path.dirname(dest_filename), exist_ok=True)
+
+            copy_file(
+                src_filename, dest_filename, verbose=self.verbose, dry_run=self.dry_run
+            )
+            if ext._needs_stub:
+                self.write_stub(package_dir or os.curdir, ext, True)
+
+
+# Allow to load dependencies from direcotry where vpi/vhpi/fli library is located on osx
+def _extra_link_args(lib_name):
+    if sys.platform == "darwin":
+        return ["-Wl,-install_name,@loader_path/%s.so" % lib_name]
+    else:
+        return []
+
+
+def _get_python_lib_link():
+    """ Get name of python library used for linking """
+
+    if sys.platform == "darwin":
+        ld_library = sysconfig.get_config_var("LIBRARY")
+    else:
+        ld_library = sysconfig.get_config_var("LDLIBRARY")
+
+    if ld_library is not None:
+        python_lib_link = os.path.splitext(ld_library)[0][3:]
+    else:
+        python_version = sysconfig.get_python_version().replace(".", "")
+        python_lib_link = "python" + python_version
+
+    return python_lib_link
+
+
+def _get_python_lib():
+    """ Get the library for embedded the python interpreter """
+
+    if os.name == "nt":
+        python_lib = _get_python_lib_link() + "." + _get_lib_ext_name()
+    else:
+        python_lib = "lib" + _get_python_lib_link() + "." + _get_lib_ext_name()
+
+    return python_lib
+
+# TODO Add "-Wconversion" to _base_warns when fixed
+_base_warns = ["-Wall", "-Wextra", "-Wcast-qual", "-Wwrite-strings"]
+_cc_warns = _base_warns + ["-Wstrict-prototypes", "-Waggregate-return"]
+_ccx_warns = _base_warns + ["-Wnon-virtual-dtor", "-Woverloaded-virtual"]
+
+_extra_cc_compile_args = ["-std=gnu99"] + _cc_warns
+_extra_cxx_compile_args = ["-std=c++11"] + _ccx_warns
```

Comment:
Thanks! cc @ktbarrett, just so you know we've backed that warning out for now. We can reinstate it as part of a PR in which CI passes.